easybuild/easyblocks/a/amber.py

```diff
@@ -0,0 +1,195 @@
+
+        # Set the AMBERHOME environment variable
+        env.setvar('AMBERHOME', self.installdir)
+        try:
+            os.chdir(self.installdir)
+        except OSError, err:
+            raise EasyBuildError("Could not chdir to {0}: {1}".format(self.installdir, err))
+
+        # Kenneth Hoste recommends making sure the LIBS env var is unset
+        env.unset_env_vars(['LIBS'])
+
+        # Set some other environment variables
+        for mathlib in ['imkl']:
+            mklroot = get_software_root(mathlib)
+            if mklroot:
+                env.setvar('MKL_HOME', mklroot)
+
+        for mpilib in ['impi', 'OpenMPI', 'MVAPICH2', 'MPICH2']:
+            mpiroot = get_software_root(mpilib)
+            if mpiroot:
+                env.setvar('MPI_HOME', mpiroot)
+
+        common_configopts = ["--no-updates", "-static", "-noX11"]
+        netcdfroot = get_software_root('netCDF')
+        if netcdfroot:
+            common_configopts.append("--with-netcdf")
+            common_configopts.append(netcdfroot)
+        pythonroot = get_software_root('Python')
+        if pythonroot:
+            common_configopts.append("--with-python")
+            common_configopts.append(os.path.join(pythonroot, 'bin', 'python'))
+
+        # If the Intel compiler is not used, we can't build CUDA Amber.
+        do_cuda = False
+        compilerstring = ''
+        if self.toolchain.comp_family() == toolchain.INTELCOMP:
+            do_cuda = True
+            compilerstring = 'intel'
+        elif self.toolchain.comp_family() == toolchain.GCC:
+            compilerstring = 'gnu'
+        else:
+            raise EasyBuildError("Don't know how to compile with compiler family {0} -- check EasyBlock?".format(self.toolchain.comp_family()))
+        
+        buildtargets = [('', 'test')]
+        if self.toolchain.options.get('usempi', None):
+            buildtargets.append(('-mpi', 'test.parallel'))
+        if do_cuda:
+            cudaroot = get_software_root('CUDA')
+            if cudaroot:
+                env.setvar('CUDA_HOME', cudaroot)
+                buildtargets.append(('-cuda', 'test.cuda'))
+
+        for flag, testrule in buildtargets:
+            # Configure
+            self.cfg['configopts'] = ' '.join(common_configopts + [flag, compilerstring])
+            self.configure_step()
+
+            # Build in situ using 'make install'
+            # Note: not "build"
+            super(EB_Amber, self).install_step()
+
+            # Test
+            self.cfg['runtest'] = testrule
+            super(EB_Amber, self).test_step()
+
+            # Clean, overruling the normal "build"
+            self.cfg['prebuildopts'] = ''
+            self.cfg['buildopts'] = 'clean'
+            super(EB_Amber, self).build_step()
+
+    def test_step(self):
+        pass
+
+    def install_step(self):
+        """In Amber, installation is conflated with building,
+        so that 'make install' is done during the build step."""
+        pass
```

Comment:
wouldn't it be better to pass the whole `build_step` body in here then?

i.e., collapse build&install into the install step, rather than the build step?